Formula/open-coarrays.rb

```diff
@@ -0,0 +1,57 @@
+class OpenCoarrays < Formula
+  desc "open-source coarray Fortran ABI, API, and compiler wrapper"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.8.2/OpenCoarrays-1.8.2.tar.gz"
+  sha256 "dc5b98a62de85e38500f431b05290415d4a4a87001c0ba03fbe57403ad62010b"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Skip build time tests (not recommended)"
+
+  depends_on :fortran
+  depends_on :mpi => [:cc, :f90]
+  depends_on "cmake" => :build
+
+  # GFortran >= 5.4 required; no other Homebrew supported compilers are compatible
+  # but the mpicc compiler may be clang (which is ABI compatible with GCC)
+  fails_with :gcc
+  fails_with :gcc_4_0
+  [("4.3".."4.9"), ("5.0".."5.3")].each do |series|
```

Comment:
> I can test this, but it might necessitate rebuilding GCC 5 from source....

Actually I can try getting it from the brew-versions tap, if you think that will work and if they provide bottles... I'll try it out and see.